sopel/coretasks.py

```diff
@@ -791,3 +792,20 @@ def track_topic(bot, trigger):
     if channel not in bot.channels:
         return
     bot.channels[channel].topic = trigger.args[-1]
+
+
+@sopel.module.rule(r'(?u).*(https?://\S+).*')
```

Comment:
I wonder about protocol support. Like, there's no reason that plugins shouldn't be able to register handlers for any kind of URL, not just HTTP(S) links, especially since your added `sopel.web.search_urls()` function includes `ftp` in the list of protocols. (Side note: `ftps` exists, too. 😸)

But like, a gaming channel might want a bot to handle `steam://` links, for example.

Could be a separate PR, sure! But I think it's worth putting in a bit of extra work on supporting arbitrary protocols if enabled by the bot owner and/or a plugin—and that means not hard-coding `https?` here.

Reply:
Hey, I wonder the same thing, because when I decided to kept the old behavior, I found it odd. At this point, I changed so many little things that I think it's safe to change a bit more.

Say I change the rule to `.*`, then perform a search based on a list of protocol provided by configuration. Say `core.url_allowed_schemes`?